File: code/ch07/7.3.1_integer_replacement.py
class Solution:
    def integerReplacement(self, n:int)->int:
        if n == 1:
            return 0
        elif n % 2 == 0:
            return 1 + self.integerReplacement(n // 2)
        else:
            return min(self.integerReplacement(n + 1), self.integerReplacement(n - 1)) + 1

File: code/ch07/7.3.1_integer_replacement.java
class Solution {
    public int integerReplacement(int n) {
        if (n == 1)
            return 0;
        if (n == Integer.MAX_VALUE)
            return 32;
        if (n % 2 == 0)
            return integerReplacement(n / 2) + 1;
        else
            return Math.min(integerReplacement(n + 1), integerReplacement(n - 1)) + 1;
    }
}